contracts/ExampleNFT-v2.cdc

```diff
@@ -0,0 +1,425 @@
+    }
+
+    /// Returns the CollectionDisplay view for the NFT type that is specified 
+    pub fun getCollectionDisplay(nftType: Type): MetadataViews.NFTCollectionDisplay? {
+        switch nftType {
+            case Type<@ExampleNFT.NFT>():
+                let media = MetadataViews.Media(
+                    file: MetadataViews.HTTPFile(
+                        url: "https://assets.website-files.com/5f6294c0c7a8cdd643b1c820/5f6294c0c7a8cda55cb1c936_Flow_Wordmark.svg"
+                    ),
+                    mediaType: "image/svg+xml"
+                )
+                return MetadataViews.NFTCollectionDisplay(
+                    name: "The Example Collection",
+                    description: "This collection is used as an example to help you develop your next Flow NFT.",
+                    externalURL: MetadataViews.ExternalURL("https://example-nft.onflow.org"),
+                    squareImage: media,
+                    bannerImage: media,
+                    socials: {
+                        "twitter": MetadataViews.ExternalURL("https://twitter.com/flow_blockchain")
+                    }
+                )
+            default:
+                return nil
+        }
+    }
+
+    /// Resource that an admin or something similar would own to be
+    /// able to mint new NFTs
+    ///
+    pub resource NFTMinter {
+
+        /// mintNFT mints a new NFT with a new ID
+        /// and returns it to the calling context
+        pub fun mintNFT(
+            name: String,
+            description: String,
+            thumbnail: String,
+            royalties: [MetadataViews.Royalty]
+        ): @ExampleNFT.NFT {
+
+            let metadata: {String: AnyStruct} = {}
+            let currentBlock = getCurrentBlock()
+            metadata["mintedBlock"] = currentBlock.height
+            metadata["mintedTime"] = currentBlock.timestamp
+            metadata["minter"] = recipient.owner!.address
+
+            // this piece of metadata will be used to show embedding rarity into a trait
+            metadata["foo"] = "bar"
+
+            // create a new NFT
+            var newNFT <- create NFT(
+                name: name,
+                description: description,
+                thumbnail: thumbnail,
+                royalties: royalties,
+                metadata: metadata,
+            )
+
+            return <-newNFT
+        }
+    }
+
+    init() {
+
+        // Set the named paths
+        self.MinterStoragePath = /storage/cadenceExampleNFTMinter
+
+        // Create a Collection resource and save it to storage
+        let collection <- create Collection()
+        let defaultStoragePath = collection.defaultStoragePath
+        let defaultPublicPath = collection.defaultPublicPath
+        self.account.save(<-collection, to: defaultStoragePath)
+
+        // create a public capability for the collection
+        self.account.link<&ExampleNFT.Collection{NonFungibleToken.CollectionPublic, MetadataViews.ResolverCollection}>(
```

Comment:
This should be linked with all restrictions? Are the other ones gone?

Reply:
yep, forgot to add `NonFungibleToken.Receiver` there